components/org.wso2.carbon.identity.data.publisher.authentication.audit/src/main/java/org/wso2/carbon/identity/data/publisher/authentication/audit/AuthenticationAuditLoggingHandler.java

```diff
@@ -271,4 +281,61 @@ private String addRemoteAddressAndAgent(String data) {
                 + "\"";
         return data;
     }
+
+    /**
+     * Returns username for audit logs based on log masking config.
+     *
+     * @param username Username.
+     * @return username. Returns masked username if log masking is enabled.
+     * */
+    private String getUsernameForAuditLog(String username) {
+
+        if (LoggerUtils.isLogMaskingEnable) {
+            return LoggerUtils.getMaskedContent(username);
+        }
+        return username;
+    }
+
+    /**
+     * Returns initiator for audit logs based on log masking config.
+     *
+     * @param username     Username.
+     * @param tenantDomain Tenant domain.
+     * @return initiator. Returns userId if log masking is enabled, if userId cannot be resolved then returns the masked
+     * username.
+     * */
+    private String getInitiator(String username, String tenantDomain) {
```

Comment:
let's return an Optional here

Reply:
This method won't return null as the initiator in any scenario.